### PR TITLE
Feature: mark duplicate egg locations

### PR DESCRIPTION
**Pull request by appable0**
## What
Marks duplicate egg locations in red when located using the Egglocator, Shared Egg Waypoints, or the Show All Waypoints feature. There is also an option to always mark nearby duplicate egg locations (within 10 blocks) in red.

<details>
<summary>Images</summary>

Screenshot 2024-05-29 at 9 53 58 PM
</details>

## Changelog New Features
+ Mark duplicate egg locations in red for unlocking some of the new rabbits. - appable
    * Option to always mark nearby duplicate egg locations.
+ Added "Collected Locations" line to the Unclaimed Eggs Display. - appable
    * This may be inaccurate if you've already collected eggs on an island.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityUniqueEggLocations.kt
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityUniqueEggLocations.kt:1]: some unused imports in here

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityUniqueEggLocations.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    fun saveNearestEgg() {
+        val location = islandEggLocations?.minByOrNull { it.distanceSqToPlayer() } ?: return
+        if (location.distanceSqToPlayer() > 100) {
+            ChatUtils.debug("""
+                Player too far from any known egg location! 
+                Player location: ${LocationUtils.playerLocation()}
+                Closest egg: $location
+            """.trimIndent())
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityUniqueEggLocations.kt]: This should probably be an ErrorManager.skyHanniError so we get data about this

File: src/main/java/at/hannibal2/skyhanni/config/features/event/HoppityEggsConfig.java [hunk 2]
```diff
@@ -44,6 +44,18 @@ public class HoppityEggsConfig {
     @FeatureToggle
     public boolean showClaimedEggs = false;
 
+    @Expose
+    @ConfigOption(name = "Highlight Duplicate Egg Locations in Red", desc = "Marks egg locations which you have already found in red.")
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/event/HoppityEggsConfig.java]: This config option name is too long, maybe just Highlight Duplicate Eggs.

File: src/main/java/at/hannibal2/skyhanni/config/features/event/HoppityEggsConfig.java [hunk 3]
```diff
@@ -44,6 +44,18 @@ public class HoppityEggsConfig {
...
+    @ConfigEditorBoolean
+    @FeatureToggle
+    public boolean highlightDuplicateEggLocations = false;
+
+    @Expose
+    @ConfigOption(name = "Show Nearby Duplicate Egg Locations", desc = "Always show duplicate eggs when nearby.")
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/event/HoppityEggsConfig.java]: Same here, the name is too long

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
Feature works completely fine, just a few things I noticed.

File: src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityUniqueEggLocations.kt
Comment (appable0) [reply src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityUniqueEggLocations.kt:1]: I'm positive I enabled optimize imports months ago! Fixed now, hopefully...

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**